### ISSUE TITLE
PH2101: False positive

### ISSUE THREAD

**Issue by ynsehoornenborg** [philips-software/roslyn-analyzers#793]
The following code raises a diagnostic, although the code has been written according to its suggestion:

``` cs
var model = aModel as SomeKindOfModel;
var oldModel = anOldModel as SomeKindOfModel;
if (oldModel != null && oldModel.IsValid) {
    ...
}            
if (model != null && model.IsValid) {
   ...
}

```

The diagnostic is thrown on the second condition, while the first is accepted.
Moving the definition of `model` to the line just before the second condition solves the issue.
